scripts/issue/333/swap_dfd_comp_for_oxd.py

```diff
@@ -53,17 +59,27 @@ def bridge_usdc_to_ftm(mantissa):
     anytoken = info['anyToken']['address']
 
     router = interface.IAnyswapV6Router(info['router'], owner=TROPS.account)
-    USDC.approve(router, mantissa)
+    USDC_ETH.approve(router, mantissa)
     router.anySwapOutUnderlying['address,address,uint256,uint256'](
         anytoken,
-        registry.ftm.badger_wallets.treasury_ops_multisig,
+        TROPS_FTM.address,
         mantissa,
         250
     )
 
     TROPS.post_safe_tx(call_trace=True)
 
 
-def swap_usdc_for_oxd():
+def swap_usdc_for_oxd(amount_mantissa, swap_all=False):
     # swap for oxd
-    pass
+    TROPS_FTM.init_solidly()
+    TROPS_FTM.take_snapshot([USDC_FTM, OXD])
+
+    TROPS_FTM.solidly.swap_tokens_for_tokens(
+        USDC_FTM,
+        USDC_FTM.balanceOf(TROPS_FTM) if swap_all else amount_mantissa,
+        [(USDC_FTM, OXD, True)]
+        )
+
+    TROPS_FTM.print_snapshot()
```

Comment:
ok i got it, so the actual path we want to pass here is: `[(USDC_FTM, WFTM, False), (WFTM, OXD, False)]`. the pair that belongs to the direct path from $usdc to $oxd does exist, but isnt deep enough. note you then have to use `tokenOut = path[-1][-2] if type(path[0]) == tuple else path[-1]` in `uni_v2.py`.

however, even better, by using `TROPS_FTM.solidly.router.getAmountOut` we can get the stable bool programmatically; ie it will return the amount and whether the stable or non stable route is the most efficient.

if we want to do it properly, i actually think we should pass the path in the traditional sense (`[USDC_FTM, WFTM, OXD]`, also keeping the signature the same) and let `Solidly.swap_tokens_for_tokens` build the list of tuples to pass to `swapExactTokensForTokens` based on `getAmountOut`.

Reply:
you are right, using `getAmountOut` returns the pool type, pushing fixes so that `univ2.swap` is more intelligent with that and the path